priam/src/main/java/com/netflix/priam/backup/AbstractFileSystem.java

```diff
@@ -221,13 +221,13 @@ public Long retriableCall() throws Exception {
 
             } catch (Exception e) {
                 backupMetrics.incrementInvalidUploads();
-                backupNotificationMgr.notify(path, UploadStatus.FAILED);
                 logger.error(
                         "Error while uploading file: {} to location: {}. Exception: Msg: [{}], Trace: {}",
                         localPath,
                         remotePath,
                         e.getMessage(),
                         e.getStackTrace());
+                backupNotificationMgr.notify(path, UploadStatus.FAILED);
```

Comment:
👍 obviates #1015
any thoughts on incorporating the change to remove `e.getStackTrace()` and instead passing `e` as the last parameter to get the stack trace?